### ISSUE TITLE
Creazione autofattura non riporta l'importo nella riga

### ISSUE THREAD

**Issue by marcus905** [devcode-it/openstamanager#1441]
## Comportamento richiesto

Alla creazione di un autofattura (caso specifico da acquisto con reverse charge estero) nell'autofattura generata, dovrebbe essere riportato l'imponibile.

## Comportamento attuale

La fattura viene creata con imponibile 0.

Testato su demo 2.4.54 https://demo.osmcloud.it/2454/editor.php?id_module=14&id_record=149

## Possibile soluzione

Non ho potuto ricercare lato sorgente dove avviene.

## Passi per riprodurre il comportamento

[Per i bug] Descrivi dettagliatamente i singoli passi per riprodurre il malfunzionamento.

1. Creazione fattura in ingresso che richiede autofattura
2. Generazione autofattura
3. Verifica imponibile

## Contesto

Inserisci le informazioni riguardanti il tuo ambiente di esecuzione. Può essere utile per individuare problemi riproducibili solo con condizioni specifiche.

* Modulo: fatture
* Versione del progetto: 2.4.54
* Versione PHP: 8.0
* Tipo di server: Aruba/Demo

**Comment by sajotrei** [comment]
sembra essere lo stesso problema segnalato 
______
**Iva in fatture non elettroniche #1435**
______
https://forum.openstamanager.com/d/2530-risolto-autofattura-integrazione-reverse-charge-interno-misto

**Comment by marcus905** [comment]
Il problema e' che la riga intera e' 0, non solo l'iva (e come tra l'altro anche quell'issue riporta.

**Comment by sajotrei** [comment]
si si valore a zero sia imponibile che iva

**Comment by marcus905** [comment]
Ho visto il mio errore, non c'e' bug per se, ma non capisco il senso di consentire la generazione di autofattura a 0. Un warning sarebbe comodo.

chiudo, nel dubbio

**Comment by marcus905** [comment]
Riapro l'issue. Il bug e' presente in presenza di fatture da esercente estero (es. wizzair/easyjet) con iva esposta ma indetraibile e senza fattura elettronica dove bisogna emettere autofattura con la stessa IVA.

**Comment by Pek5892** [comment]
Buongiorno @marcus905,
la fattura di acquisto estera andrà registrata con aliquota non imponibile, e successivamente si dovrà generare e inviare l'autofattura tra le vendite e registrare l'autofattura tra gli acquisti una volta ricevuta.
Vi lascio il link alla documentazione
https://docs.openstamanager.com/v/2.4.54/guide/esempi/autofattura

**Comment by marcus905** [comment]
Esatto, ma nella generazione dell'auto fattura dal menù crea, l'imponibile e l'imposta rimangono 0. Dovrebbero popolarsi da soli.

**Comment by sajotrei** [comment]
> Buongiorno @marcus905, la fattura di acquisto estera andrà registrata con aliquota non imponibile, e successivamente si dovrà generare e inviare l'autofattura tra le vendite e registrare l'autofattura tra gli acquisti una volta ricevuta. Vi lascio il link alla documentazione https://docs.openstamanager.com/v/2.4.54/guide/esempi/autofattura


ciao Valentina, si si abbiamo già visto la documentazione.

ma come avevamo indicato nella 1435 e come ha ribadito @marcus905, i **valori rimangono a zero non vengono valorizzati.**

**Comment by Pek5892** [comment]
Nella fattura indicata in demo l'autofattura viene generata con righe a zero perchè la riga presente nella fattura di acquisto registrata **non riporta una delle aliquote con natura iva tra quelle previste per il calcolo dell'iva compensativa**.

Questo automatismo è stato introdotto, come già spiegato nell'issue linkata, per consentire la registrazione di fatture in regime di reverse charge misto (-> https://forum.openstamanager.com/d/2530-risolto-autofattura-integrazione-reverse-charge-interno-misto/2) dove quindi in un documento siano presenti righe su cui andare a calcolare l'IVA integrativa, e righe in cui l'iva sia invece già presente e non da integrare. 

Le righe della fattura di acquisto ricevuta dovrà quindi essere registrata con un'aliquota con natura IVA compresa tra ['N2', 'N2.1', 'N2.2', 'N3', 'N3.1', 'N3.2', 'N3.3', 'N3.4', 'N3.5', 'N3.6', 'N6', 'N6.1', 'N6.2', 'N6.3', 'N6.4', 'N6.5', 'N6.6', 'N6.7', 'N6.8', 'N6.9'] per far si che venga calcolata l'IVA su queste righe. 
**Tutte le righe con natura IVA diversa da questa non verranno considerate per il calcolo dell'IVA compensativa presente nell'autofattura generata**.

Ricordiamo inoltre che la fattura di acquisto che viene registrata corrisponde alla fattura estera ricevuta, **non all'autofattura**, che andrà invece importata tra gli acquisti a seguito dell'invio e accettazione dello SDI dell'autofattura generata tra le vendite.

**Comment by marcus905** [comment]
Che il comportamento sia questo mi e' chiaro. Il problema è che l'autofattura, in caso di iva indetraibile, la deve riportare anche! [(correggetemi se sbaglio)](https://www.apindustria.vi.it/images/articoli_web/FISCALE/11_10_Acquisto-viaggi-aerei-tramite-internetosservazioni-integrative.pdf) pag 2 punto 3.

Se ho ragione, il comportamento del gestionale e' errato! perche in caso di iva indetraibile il comportamento è identico al reverse charge.

**Comment by marcus905** [comment]
@Pek5892 posso avere input su cosa fare qui? Perche posso fare un fix e annessa pullrequest, ma voglio capire se sto sbagliando io nella gestione contabile. Grazie

**Comment by Pek5892** [comment]
Buongiorno @marcus905 ,
ci siamo confrontati: probabilmente serve che nell'autofattura generata ci sia lo stesso importo e anche la stessa aliquota IVA presente nella fattura di acquisto originale, abbiamo capito bene?

**Comment by marcus905** [comment]
Buongiorno, al meglio delle mie conoscenze in materia, come da PDF linkato su a corroborarle... si.
